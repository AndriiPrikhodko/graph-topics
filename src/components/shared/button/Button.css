.standard-button {
    background-color: #666;
    color: #fff;
    border: none;
    font-size: 14pt;
    margin-top: 1px;
    font-family: math;
    cursor: pointer;
    border-radius: 4px;
    transition-duration: 0.4s;
    box-shadow: 0 0 2px black;
}
  
.standard-button:hover {
    background-color: #e6e6e6;
    color: #333;
    outline-color: #adadad;
}