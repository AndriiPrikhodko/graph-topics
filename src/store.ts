import { configureStore } from '@reduxjs/toolkit'
import rootReducer from './reducers'
import thunk from 'redux-thunk'

const store = configureStore({
  reducer: {
    graphData: rootReducer.graphSlice,
    graphList: rootReducer.graphListSlice
  },
  middleware: [thunk],
})

export default store;