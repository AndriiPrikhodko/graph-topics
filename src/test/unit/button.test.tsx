import { vi, expect } from 'vitest'
import { test } from 'vitest'
import { render } from '@testing-library/react'
import store from '../../store'
import { Provider } from 'react-redux'
import Button from '../../components/shared/button'
import userEvent from '@testing-library/user-event'

test('shared button should invoke handle on click', async () => {

    const Props = {
        label: 'Test',
        'testid': 'button',
        onClick: vi.fn(),
    }

    const ButtonR = render(
        <Provider store={store}>
            <Button {...Props}/>
        </Provider>
    )

    const button = ButtonR.getByTestId('button')
    await userEvent.click(button)

    expect(Props.onClick).toHaveBeenCalled()
    expect(button).toMatchSnapshot()
})