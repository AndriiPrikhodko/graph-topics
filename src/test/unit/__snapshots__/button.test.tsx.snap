// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shared button should invoke handle on click 1`] = `
<button
  class="standard-button"
  data-testid="button"
>
  Test
</button>
`;
